toxcore/tox_private.c

```diff
@@ -0,0 +1,142 @@
+/* SPDX-License-Identifier: GPL-3.0-or-later
+ * Copyright © 2016-2022 The TokTok team.
+ * Copyright © 2013 Tox project.
+ */
+
+/**
+ * The Tox private API (for tests).
+ */
+#include "tox_private.h"
+
+#include <assert.h>
+
+#include "ccompat.h"
+#include "tox_struct.h"
+
+#define SET_ERROR_PARAMETER(param, x) \
+    do {                              \
+        if (param != nullptr) {       \
+            *param = x;               \
+        }                             \
+    } while (0)
+
+const Tox_System tox_default_system = {
+    nullptr,  // mono_time_callback
+    nullptr,  // mono_time_user_data
+};
+
+void tox_lock(const Tox *tox)
```

Comment:
I see lock and unlock, but no init or deinit, missing or not needed?

Reply:
This was moved out of tox.c. I really don't want to touch the pthread stuff in this PR.

Reply:
clarified in IRC